src/helpers.jl

```diff
@@ -63,15 +63,14 @@ function AprilTagDetector(tagfamily::TagFamilies)
     apriltag_detector_add_family(td, tf)
 
     return AprilTagDetector(td,tf)
-
 end
 
 
 """
 	AprilTagDetector(img)
 Run the april tag detector on a image
 """
-function (detector::AprilTagDetector)(image)::Vector{AprilTag}
+function (detector::AprilTagDetector)(image::Array{ColorTypes.Gray{T}, 2}) where T
```

Comment:
#### Julia related question
@GearsAD  and @dehann 
Is there a way to call this for all Array of 8 bit types (UInt8, N0f8, Int8, Gray{N0f8}, etc)
function (detector::AprilTagDetector)(image::Array{T, 2}) where **"T has 8 bit storage type"**
I basically want to call it on any type that will work with 
```reinterpret(UInt8, image)```

Reply:
Hi @Affie , yes, I think we should do something like:
```julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.2 (2017-12-13 18:08 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using FixedPointNumbers

julia> const EightBitTypes = Union{Int8, UInt8, N0f8}
Union{FixedPointNumbers.Normed{UInt8,8}, Int8, UInt8}

julia> function ff(x::EightBitTypes)
         @show "this happened, ", x
         nothing
       end
ff (generic function with 1 method)

julia> ff(Int8(1))
("this happened, ", x) = ("this happened, ", 1)

julia> ff(N0f8(1))
("this happened, ", x) = ("this happened, ", 1.0N0f8)

julia> ff(1)
ERROR: MethodError: no method matching ff(::Int64)
Closest candidates are:
  ff(::Union{FixedPointNumbers.Normed{UInt8,8}, Int8, UInt8}) at REPL[3]:2
```

Reply:
Its a little harder than that sorry.  We would need either 
```julia
Union{Array{Gray{Int8},2}, Array{Gray{UInt8},2}, ...}
``` 
or somehow do inheritance from `<: EightBitTypes` type, although not sure how to do the second yet.

Reply:
Thanks, I will look at the union thing. 
I reckon for the AprilTags application it should be only unsigned 8 bit types.
The other thing is gray can also be an float type.

Reply:
Multiple dispatch for the win.